### PR TITLE
Sidebar menu

### PR DESCRIPTION
**Pull request by duranmla**
**What:**
Allow to toggle sidebar menu component

**Why:**
closes https://app.asana.com/0/1168997577035609/1199650841826371
closes https://app.asana.com/0/1168997577035609/1199650841826381
closes https://app.asana.com/0/1168997577035609/1199650841826378

**How:**
- Add a new setting variable
- Update layout to fit sidebar
- Create a new component to render sidebar

**Media:**
https://www.loom.com/share/d0673dd7c2a04deca2f3b6666536e5a6

**Extra:**
- community self navigation to avoid reload a996f04
- restore admin dropdown f3c1dc6
- allow to use legacy decorators and proper lint check b12fe0d

_Overall minor details check. You can take a look to first commits on this Pull Request_

### CODE REVIEW

**Comment by duranmla** [comment]
_After video I realise that currently if you go to "all categories" after expand a category with sub categories this later one won't be collapsed, I need to fix that_

File: javascripts/discourse/components/dc-sidebar-subcategories.js.es6 [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  currentCategoryId: null,
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+    this._computeAttrs();
+  },
+
+  _computeAttrs() {
+    // Avoid to do any comparation unless there is value
+    if (!this.currentCategoryId) return;
```
Comment (orlando) [inline javascripts/discourse/components/dc-sidebar-subcategories.js.es6:21]: why we cannot check this anyway below and return false?

File: scss/templates/user.scss [hunk 2]
```diff
@@ -204,6 +204,7 @@
 
   @include media-breakpoint-down(sm) {
     ul {
+      display: -ms-grid;
```
Comment (orlando) [inline scss/templates/user.scss:207]: do we need to put prefixes here? I think this should be handled by autoprefixer right?

File: settings.yml [hunk 3]
```diff
@@ -15,15 +15,28 @@ header_categories_custom_links:
...
   type: bool
   default: true
+  description:
+    en: "Show message banner on community homepage"
 show_current_efforts:
   type: bool
   default: false
+  description:
+    en: "make visible pinned topics over 'all categories' page"
+show_sidebar:
```
Comment (orlando) [inline settings.yml:25]: ✔️

**Review by orlando — COMMENTED** [review]
Just a few comments, but LGTM

File: scss/templates/user.scss (hunk 2 above)
Comment (duranmla) [reply scss/templates/user.scss:207]: Yes we should, and this actually remind me to create the ticket. 
_I am having a extension checking upon a11y matters and it include some CSS rules depending upon supported browsers._

https://app.asana.com/0/1159164196409129/1199915290849534

File: javascripts/discourse/components/dc-sidebar-subcategories.js.es6 (hunk 1 above)
Comment (duranmla) [reply javascripts/discourse/components/dc-sidebar-subcategories.js.es6:21]: Thanks, this actually inject the bug I was having when going towards "all categories" and not collide the already expanded parent category

**Comment by duranmla** [comment]
This pull request has been updated already with all the feedback given.
_Please feel free to take a look and merge_

**Review by orlando — APPROVED** [review]
LGTM!

**Pull request merged**